spread-shellcheck

```diff
@@ -133,15 +140,74 @@ def checksection(data, env: Dict[str, str]):
                             stdout=subprocess.PIPE,
                             stdin=subprocess.PIPE,
                             shell=True)
-    stdout, _ = proc.communicate(input='\n'.join(script_data).encode('utf-8'), timeout=30)
+    stdout, _ = proc.communicate(input='\n'.join(script_data).encode('utf-8'), timeout=60)
     if proc.returncode != 0:
         raise ShellcheckRunError(stdout)
 
 
+class Cacher:
+    _instance = None
+    def __new__(cls):
+        if cls._instance is not None:
+            return cls._instance
+        instance = super().__new__(cls)
+        cls._instance = instance
+        instance._enabled = True
+        instance._lock = Lock()
+        instance._hit =0
+        instance._miss = 0
+        return instance
+
+    def disable(self):
+        logging.debug("caching is disabled")
+        self._enabled = False
+
+    @staticmethod
+    def _cache_path_for(digest):
+        prefix = digest[:2]
+        return Path.home().joinpath(".cache", "spread-shellcheck", prefix, digest)
+
+    def is_cached(self, data, path):
```

Comment:
Is "path" used in this func?

Reply:
It's not, I was thinking whether we should maybe include the path in the hash too. I can drop it from the arguments and we can add it back later if needed.

Reply:
FWIW, maybe it's also worth adding shellchec version to the digest, so that e get a miss when shellcheck is updated.

Reply:
I see - I personally don't see the value of path but maybe I'm missing something? The shellcheck version seems sensible but that could be a followup.